### PR TITLE
[GEOS-10505] Display style format as new column in Styles-list, along with Style Name and Workspace

### PR DESCRIPTION
**Pull request by mmohiahmed**
[GEOS-10505](https://osgeo-org.atlassian.net/browse/GEOS-10505)

It's beneficial for our company (and hopefully other users), to add a new column 'Format' in Data > Styles table along with 'Name' and 'Workspace'. Formats will be SLD, ZIP, Property etc. Will you accept this as my first OSS contribution?
# Checklist

- [X] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [X] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [X] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [X] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [X] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [X] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [X] Bug fixes and small new features are presented as a single commit.
- [X] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by petersmythe** [comment]
@aaime @jodygarnett or other core developers, please take the time to review my AfriGIS colleague's PR.  We're trying to build up some internal developer capacity to tackle small changes, and hopefully more in time.  Thanks, Peter

**Comment by jodygarnett** [comment]
@petersmythe I thought I had already provided feedback, here is a doc failure, and I am not sure how to add a test case to this one. Perhaps copy one of the other examples and confirm that the format is visible on screen?

**Comment by aaime** [comment]
Should not be merged without a test. 

In order to add a test:
* Extend the existing StylePageTest
* Add one SLD 1.1 style (to have something different)
* Start StylePage, check the column is there and has the right contents (see the testWorkspaceFilter method for inspiration, you could have the style names set up so that only one style shows at a time, and check the SLD 1.0 and SLD 1.1 cases one by one.)

File: src/web/core/src/main/java/org/geoserver/web/wicket/StyleFormatLabel.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        if (model.getObject() != null) {
+            String format = (String) model.getObject();
+            formatDisplayName = Styles.handler(format).getName();
+
+            if (versionString != null) {
+                formatDisplayName =
+                        new StringBuilder(formatDisplayName)
+                                .append(" ")
+                                .append(versionString)
+                                .toString();
```
Comment (aaime) [inline src/web/core/src/main/java/org/geoserver/web/wicket/StyleFormatLabel.java:34]: [No need for StringBuilder for simple concatenation](https://dzone.com/articles/string-concatenation-performacne-improvement-in-ja). Use it in loops or string builds spanning several lines, but here it's overkill and harder to read.

```suggestion
                        new formatDisplayName + " " + versionString;
```

File: src/web/wms/src/main/java/org/geoserver/wms/web/data/StylePage.java [hunk 2]
```diff
@@ -63,6 +65,18 @@ protected Component getComponentForProperty(
                                             id,
                                             StyleProvider.CREATED_TIMESTAMP.getModel(itemModel));
                                 }
+                                if (property == StyleProvider.FORMAT) {
+                                    Version formatVersion =
```
Comment (aaime) [inline src/web/wms/src/main/java/org/geoserver/wms/web/data/StylePage.java:69]: Setting up the label is split among the code here and the decated StyleFormatLabel component. Please move everything inside StyleFormatLabel instead, you could give it the entire itemModel as constructor argument for example, and perform extraction and formatting in there.

File: src/web/wms/src/main/java/org/geoserver/wms/web/data/StylePage.java [hunk 3]
```diff
@@ -151,4 +165,18 @@ protected static boolean isDefaultStyle(CatalogInfo catalogInfo) {
...
+        Comparable<?> major = formatVersion.getMajor();
+        Comparable<?> minor = formatVersion.getMinor();
+
+        if (major == null) {
+            return null;
+        }
+
+        if (minor == null) return major.toString();
+
+        return new StringBuilder(major.toString()).append(".").append(minor.toString()).toString();
```
Comment (aaime) [inline src/web/wms/src/main/java/org/geoserver/wms/web/data/StylePage.java:180]: Same as above, no need to use StringBuilder.

**Review by aaime — CHANGES_REQUESTED** [review]

**Comment by aaime** [comment]
Regarding the two build failures:
* The docs one is related to a new warning emitted by the latest version of Sphinx. If you rebase you branch on main (or merge) you'll get the fix required (just merged):
* The checkstyle violation is related to excessive code nesting, can normally be solved by factoring out some code bits in separate methods:

```
2022-05-30T10:42:14.9443610Z 10:42:14,943 [INFO] There is 1 error reported by Checkstyle 8.29 with /home/runner/work/geoserver/geoserver/src/../build/qa/checkstyle.xml ruleset.
2022-05-30T10:42:14.9453180Z 10:42:14,944 [ERROR] src/main/java/org/geoserver/wms/web/data/StylePage.java:[73] (regexp) RegexpSingleline: Excessive nesting found. Please try to factor out the deeply nested code in a separate method or class
```

**Comment by aaime** [comment]
I've tried it locally again, found it would not display the style list anymore... this was due to some styles I have, that did not have their style handler plugged in (e..g, having CSS styles in the data dir, without the CSS plugin installed).
Before it was failing to use the styles, but would list them without issues: it's a regression, I've fixed it, merged with the latest main, and pushed the changes up. 

The build is now ongoing again, let's see if it gets to green.

**Comment by aaime** [comment]
Cool, build was green, I've squash-merged the changes.

**Pull request merged**